src/pool/srv_pool_check.c

```diff
@@ -162,6 +164,41 @@ ds_pool_clue_init(uuid_t uuid, enum ds_pool_dir dir, struct ds_pool_clue *clue)
 		goto out_path;
 	}
 
+	D_ALLOC_ARRAY(clue->pc_tgt_status, dss_tgt_nr);
+	if (clue->pc_tgt_status == NULL) {
+		D_ERROR(DF_UUIDF": failed to allocate service clue for shards status\n",
+			DP_UUID(uuid));
+		D_GOTO(out_path, rc = -DER_NOMEM);
+	}
+
+	for (i = 0; i < dss_tgt_nr; i++) {
+		rc = ds_mgmt_tgt_file(uuid, VOS_FILE, &i, &file);
+		if (file == NULL) {
+			D_ERROR(DF_UUIDF": failed to allocate file name for shards status %d\n",
+				DP_UUID(uuid), i);
+			D_GOTO(out_path, rc = -DER_NOMEM);
```

Comment:
Do we leak clue->pc_tgt_status on the error paths?

Reply:
Right, I will refresh the patch to fix it.